beamer/agent.py

```diff
@@ -36,103 +39,166 @@ def __init__(self, config: Config):
         self._config = config
         self._stopped = threading.Event()
         self._stopped.set()
+        self._cache: dict[str, Any] = {}
```

Comment:
it is just simple like this as i dont need to pass parameters to other things

Reply:
> it is just simple like this as i dont need to pass parameters to other things

It may be simpler in terms of fewer arguments being passed around, but it makes it harder to reason about and understand the code and makes it harder to reason about the lifetime of those objects. It also prevents the garbage collector from collecting those objects for the duration of the agent's life. Since, as far as I understand, we only need those objects during initialization, we should not retain them for the entire life of the agent.